src/Compilers/Core/CodeAnalysisTest/Collections/HashSet/ISet_Generic_Tests`1.cs

```diff
@@ -590,24 +640,27 @@ public void ISet_Generic_UnionWith_LargeSet()
         [MemberData(nameof(EnumerableTestData))]
         public void ISet_Generic_SymmetricExceptWith_AfterRemovingElements(EnumerableType enumerableType, int setLength, int enumerableLength, int numberOfMatchingElements, int numberOfDuplicateElements)
         {
-            ISet<T> set = GenericISetFactory(setLength);
-            T value = CreateT(532);
-            set.Add(value);
-            set.Remove(value);
-            IEnumerable<T> enumerable = CreateEnumerable(enumerableType, set, enumerableLength, numberOfMatchingElements, numberOfDuplicateElements);
-            Debug.Assert(enumerable != null);
+            if (!IsReadOnly)
+            {
+                ISet<T> set = GenericISetFactory(setLength);
+                T value = CreateT(532);
+                set.Add(value);
+                set.Remove(value);
+                IEnumerable<T> enumerable = CreateEnumerable(enumerableType, set, enumerableLength, numberOfMatchingElements, numberOfDuplicateElements);
+                Debug.Assert(enumerable != null);
 
-            IEqualityComparer<T> comparer = GetIEqualityComparer();
-            HashSet<T> expected = new HashSet<T>(comparer);
-            foreach (T element in enumerable)
-                if (!set.Contains(element, comparer))
-                    expected.Add(element);
-            foreach (T element in set)
-                if (!enumerable.Contains(element, comparer))
-                    expected.Add(element);
-            set.SymmetricExceptWith(enumerable);
-            Assert.Equal(expected.Count, set.Count);
-            Assert.True(expected.SetEquals(set));
+                IEqualityComparer<T> comparer = GetIEqualityComparer();
+                HashSet<T> expected = new HashSet<T>(comparer);
```

Comment:
Should this be changed to SegmentedHashSet? Applies to other tests in this file.

Reply:
➡️ No, these tests are verifying our implementation against a known good implementation